packages/dmn-editor/tests/e2e/mutations/addEdgeWaypoint.spec.ts

```diff
@@ -0,0 +1,134 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { expect } from "@playwright/test";
+import { test } from "../__fixtures__/base";
+import { DefaultNodeName, NodeType } from "../__fixtures__/nodes";
+import { EdgeType } from "../__fixtures__/edges";
+import { TestAnnotations } from "@kie-tools/playwright-base/annotations";
+
+test.beforeEach(async ({ editor }) => {
+  await editor.open();
+});
+
+test.describe("MUTATIONS - Add edge waypoint", () => {
+  test("Information Requirement", async ({ diagram, palette, nodes, edges }) => {
+    await palette.dragNewNode({ type: NodeType.INPUT_DATA, targetPosition: { x: 100, y: 100 } });
+    test.info().annotations.push({
+      type: TestAnnotations.WORKAROUND_DUE_TO,
+      description: "",
+    });
+    await diagram.resetFocus();
+
+    await palette.dragNewNode({ type: NodeType.DECISION, targetPosition: { x: 100, y: 300 } });
+
+    await diagram.resetFocus();
+
+    await nodes.dragNewConnectedEdge({
+      type: EdgeType.INFORMATION_REQUIREMENT,
+      from: DefaultNodeName.INPUT_DATA,
+      to: DefaultNodeName.DECISION,
+    });
+
+    await edges.addWaypoint({ from: DefaultNodeName.INPUT_DATA, to: DefaultNodeName.DECISION });
+
+    await nodes.get({ name: DefaultNodeName.DECISION }).dragTo(diagram.get(), { targetPosition: { x: 300, y: 300 } });
+
+    await expect(diagram.get()).toHaveScreenshot();
+  });
```

Comment:
There are two comments I would like to bring into our attention.

1. `test.info().annotations....` - this is something I intentionally copied exactly as it is in `addEdge.spec.ts`. I think I understand we want to warn about the need of `diagram.resetFocus()`, however are the combination of `WORKAROUND_DUE_TO` and empty `description` makes me not sure, if we are currently in some temporary state and descriptions needs to be added, or this is some special case, when we intentionally do not provide description.
2. the use of `diagram.ResetFocus()` I think it is a valid method in our fixtures because also user needs sometime reset selection/focus, my question is, if our strategy is to keep it as standalone fixture and user need to call it explicitly in the test method body, or we should try to use it internally on other fixture method, like in `dragNewNode` that could start with similar?